setup.sh

```diff
@@ -84,601 +120,184 @@ vercomp () {
-      ;;
-    esac
-  done
+  # Check of vm.max_map_count is at least $MIN_MAX_MAP_COUNT
+  VM_MAX_MAP_COUNT=$(sysctl -n vm.max_map_count)
+  if [ -z "${VM_MAX_MAP_COUNT}" ] || [ ${VM_MAX_MAP_COUNT} -lt $MIN_MAX_MAP_COUNT ]; then
+    (>&2 echo)
+    (>&2 echo $RED"The current value of the kernel configuration variable vm.max_map_count (${VM_MAX_MAP_COUNT})")
+    (>&2 echo "is lower than the required one ($MIN_MAX_MAP_COUNT+).")
+    (>&2 echo "In order to make ElasticSearch working please set it by using on root: (more at https://www.elastic.co/guide/en/elasticsearch/reference/5.x/vm-max-map-count.html)")
+    (>&2 echo $BLUE$BOLD"sysctl -w vm.max_map_count=$MIN_MAX_MAP_COUNT")
+    (>&2 echo $RED"If you want to persist it please edit the $BLUE$BOLD/etc/sysctl.conf$NORMAL$RED file")
+    (>&2 echo "and add $BLUE$BOLD vm.max_map_count=$MIN_MAX_MAP_COUNT$NORMAL$RED in it."$NORMAL)
+    $KUZZLE_PUSH_ANALYTICS'{"type": "wrong-max_map_count", "uid": "'$ANALYTICS_UUID'", "os": "'$OS'"}' $ANALYTICS_URL &> /dev/null    
+    exit $MIN_MAX_MAP_COUNT_MISMATCH
+  fi
 }
 
-collectPersonalData() {
-  echo
-  while [[ "$purpose" != [012345] ]]
-  do
-    writeBold "[❓] What are you going to use Kuzzle for?"
-    write     "    1) IoT"
-    write     "    2) Web"
-    write     "    3) Mobile"
-    write     "    4) Machine-to-machine"
-    write     "    5) Other"
-    write     "    0) Stop bugging me"
-    echo -n "> "
-    read purpose trash
-  done
-
-  case "$purpose" in
-    0) purpose="Stop bugging me" ;;
-    1) purpose="IoT" ;;
-    2) purpose="Web" ;;
-    3) purpose="Mobile" ;;
-    4) purpose="Machine-to-machine" ;;
-    5) purpose="Other" ;;
-  esac
-  echo
-  writeBold "[❓] Would you like us to reach you to have your feedback on Kuzzle? (y/N)"
-  write     "    We will be really discreet (and this will help us a lot improving Kuzzle)"
-  echo -n "> "
-  while [[ "$agreeOnPersonalData" != [yYnN] ]]
-  do
-    read agreeOnPersonalData trash
-    case "$agreeOnPersonalData" in
-      [yY])
-        echo
-        writeBold "[❓] What's your email address?${NORMAL} (press Enter to skip)"
-        echo -n "> "
-        read email trash
-        echo
-        writeBold "[❓] What's your full name?${NORMAL} (press Enter to skip)"
-        echo -n "> "
-        read name
-        echo
-        writeBold "$GREEN" "[✔] Thanks a lot!"
-        echo
-        ;;
-      [nN] | '')
-        writeBold "$BLUE" "Ok."
-        agreeOnPersonalData="n"
-        echo
-        ;;
-      *)
-        writeBold "$RED" "[✖] Please, answer Y or N."
-        ;;
-    esac
-  done
-  $DL_BIN $UL_OPTS '{"type": "collected-data", "uid": "'$UUID'", "email": "'$email'", "name": "'"$name"'", "purpose": "'"$purpose"'", "os": "'$CURRENT_OS'"}' $ANALYTICS_URL &> /dev/null
+download_docker_compose_yml() {
+  local RETRY=0
+
+  echo 
+  echo $BLUE"Downloading Kuzzle docker-compose.yml file..."$NORMAL
+
+  TEST=`eval "$KUZZLE_CHECK_DOCKER_COMPOSE_YML_HTTP_STATUS_CODE"`
```

Comment:
do we need both the eval and the the command substitution?
Further, I'd rather use `$()` than backticks

Reply:
We need the eval because our command is a string and if we do not use $() or `` then for instance: wget --server-response .... : no such file or directory
I'm curious on why you prefer $() over ``